pint_pandas/pint_array.py

```diff
@@ -512,22 +527,17 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
                 raise ValueError(
                     "Cannot infer dtype. No dtype specified and empty array"
                 )
-        if dtype is None and not isinstance(master_scalar, _Quantity):
-            raise ValueError("No dtype specified and not a sequence of quantities")
-        if dtype is None and isinstance(master_scalar, _Quantity):
+        if dtype is None:
+            if not isinstance(master_scalar, _Quantity):
+                raise ValueError("No dtype specified and not a sequence of quantities")
             dtype = PintType(master_scalar.units)
 
-        def quantify_nan(item):
```

Comment:
I'm suprised you can remove this. I'll need to check whether its needed now

Reply:
I didn't really remove anything, I just reorganized the conditional logic so that we don't test dtype against None once for the negative branch and once for the positive branch.

Reply:
nice to know, thanks for explaining